packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -597,13 +602,18 @@ export function validateBlobsAndTrees(snapshot: IOdspSnapshot): void {
 	);
 }
 
-function countTreesInSnapshotTree(snapshotTree: ISnapshotTree): number {
-	let numTrees = 0;
-	for (const [_, tree] of Object.entries(snapshotTree.trees)) {
-		numTrees += 1;
-		numTrees += countTreesInSnapshotTree(tree);
+function getTreeStatsCore(snapshotTree: ISnapshotTree, stats: ITreeStats): void {
```

Comment:
Nit: suggestion to place the function next to the getTreeStats above since they are related. 

Or better yet, we can move these tree stat calculating functions to a different file or `odspUtils`